### ISSUE TITLE
[tqdm.rich] latest rich breaks the example

### ISSUE THREAD

**Issue by andrzejnovak** [tqdm/tqdm#1306]
The example doesn't finish after updating rich

![image](https://user-images.githubusercontent.com/13226500/157205218-19755719-884b-4653-af8f-d7cc76d3be88.png)

```
rich == 11.2.0
tqdm ==  4.63.0
```

**Comment by glensc** [comment]
interesting is that i updated my test to also include `sleep(0.1)` and now the progressbar increases!.

so the rich progressbar somehow related how often it is updated, so it omits refresh if walked too fast. i guess and lacks final repaint?

**Comment by glensc** [comment]
@andrzejnovak do you have info in what rich version it worked?

**Comment by glensc** [comment]
I think this is a duplicate of the same problem: 
- https://github.com/tqdm/tqdm/issues/1266

**Comment by glensc** [comment]
I've downgraded rich as down as rich==9.6.0,

I could not downgrade more as that's the version Progress.TimeElapsedColumn was first added:
- https://github.com/Textualize/rich/releases/tag/v9.6.0

and the problem is still there.

**Comment by glensc** [comment]
I've also downgraded tqdm down as tqdm==4.56.0

I could not downgrade more as that's the version tqdm.rich was added:
- https://github.com/tqdm/tqdm/releases/tag/v4.56.0

so, the issue title is incorrect, this has never worked propertly, not just "latest" version.

**Comment by glensc** [comment]
seems to me that `def display()` is not called by tqdm if the sleep is removed

- https://github.com/tqdm/tqdm/blob/6791e8c5b3d6c30bdd2060c346996bfb5a6f10d1/tqdm/rich.py#L128-L131

I added `print("display")` into the method, and it's not printed at all if the "loop goes to fast".

**Comment by glensc** [comment]
This fixed problem for me:
- https://github.com/tqdm/tqdm/pull/1395

**Comment by kkovary** [comment]
@glensc that change fixed the problem for me too. Would be great to get #1395 merged 🤞

**Comment by glensc** [comment]
https://github.com/tqdm/tqdm/pull/1395 merged, this can be closed

**Comment by ma-sadeghi** [comment]
@glensc I still have this problem, although in VS Code interactive Window. I'd appreciate it if you could try it out on VS Code:

```python
import time
from tqdm.rich import trange, tqdm
for i in trange(10):
    time.sleep(0.2)
```

**Comment by glensc** [comment]
@ma-sadeghi did you update to 4.66.4?

**Comment by ma-sadeghi** [comment]
Yes, `tqdm.__version__` shows 4.66.4.

**Comment by glensc** [comment]
works for me from terminal

```
$ python test_tqdm.py
/lib/python3.8/site-packages/tqdm/rich.py:146: TqdmExperimentalWarning: rich is experimental/alpha
  return tqdm_rich(range(*args), **kwargs)
i=0
i=1
i=2
i=3
i=4
i=5
i=6
i=7
i=8
i=9
 100% ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10/10  [ 0:00:01 < 0:00:00 , 9 it/s ]
finished
$
```

<details><summary>Source</summary>

```python
# https://github.com/tqdm/tqdm/issues/1306#issuecomment-2092369446
from tqdm.rich import trange, tqdm
import time

it = trange(10)
for i in it:
    print(f"i={i}")
    time.sleep(0.1)

print("finished")

```

</details>

**Comment by ma-sadeghi** [comment]
Right, but as I mentioned, it doesn't work in the Interactive Window. Ctrl shift P, then look for Interactive Window, then run the snippet there

**Comment by glensc** [comment]
looks good to me

<img width="1131" alt="image" src="https://github.com/tqdm/tqdm/assets/199095/9d311cce-ecc4-4d2c-a41e-18a7f74dc2a6">


imho what you're experimenting is a different issue. probably in vscode or terminal or os.

when creating new bugreport include all details of your system, os/terminal ($TERM), etc. maybe include a screenshot or video if can't present as text.

**Comment by ma-sadeghi** [comment]
There you go: 

https://github.com/tqdm/tqdm/assets/14086031/89ebd047-3b2d-4226-8b1a-e037e74086bd

**Comment by glensc** [comment]
so, that's two users, previous user didn't provide reproducer (if you read that link I shared). so submit pr with a fix and fingers crossed that maintainer will accept it soon :)

**Comment by ma-sadeghi** [comment]
@glensc Right, I'll submit a PR soon. Thanks for helping figure out the root cause :-)
